generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs

```diff
@@ -33,65 +33,56 @@ if (enableTranslation) {
-    if (relationship.relationshipRequired) {
-        openBlockComment = `/*`;
-        closeBlockComment = `*/`;
-        break;
-    }
+  if (relationship.relationshipRequired) {
+    openBlockComment = `/*`;
+    closeBlockComment = `*/`;
+    break;
+  }
 } _%>
- <%_ if (fieldsContainBlobOrImage) { _%>
+<%_ if (fieldsContainBlobOrImage) { _%>
 const fileToUpload = '../../../../../../<%= entityParentPathAddition %><%= CLIENT_MAIN_SRC_DIR %>content/images/logo-jhipster.png';
 const absolutePath = path.resolve(__dirname, fileToUpload);
 <%_ } _%>
 export default class <%= entityClass %>UpdatePage {
     pageTitle: ElementFinder = element(by.id('<%= i18nKeyPrefix %>.home.createOrEditLabel'));
     saveButton: ElementFinder = element(by.id('save-entity'));
     cancelButton: ElementFinder = element(by.id('cancel-save'));
-    <%_ fields.filter(field => !field.id).forEach((field) => {
-            const fieldName = field.fieldName;
-            const fieldType = field.fieldType;
-            const fieldIsEnum = field.fieldIsEnum;
-            const fieldTypeBlobContent = field.fieldTypeBlobContent;
-    _%>
-    <%_ if (fieldIsEnum) { _%>
+<%_ fields.filter(field => !field.id).forEach((field) => {
+        const fieldName = field.fieldName;
+        const fieldIsEnum = field.fieldIsEnum;
+_%>
+  <%_ if (fieldIsEnum) { _%>
     <%= fieldName %>Select: ElementFinder = element(by.css('select#<%= entityFileName %>-<%= fieldName %>'));
-    <%_ } else if (['byte[]', 'ByteBuffer'].includes(fieldType) && fieldTypeBlobContent === 'text') { _%>
+  <%_ } else if (field.fieldTypeBinary && field.blobContentTypeText) { _%>
     <%= fieldName %>Input: ElementFinder = element(by.css('textarea#<%= entityFileName %>-<%= fieldName %>'));
-    <%_ } else if (['byte[]', 'ByteBuffer'].includes(fieldType)) { _%>
+  <%_ } else if (field.fieldTypeBinary) { _%>
     <%= fieldName %>Input: ElementFinder = element(by.css('input#file_<%= fieldName %>'));
-    <%_ } else { _%>
+  <%_ } else { _%>
     <%= fieldName %>Input: ElementFinder = element(by.css('input#<%= entityFileName %>-<%= fieldName %>'));
-    <%_ } _%>
-    <%_ }); _%>
-    <%_ relationships.forEach((relationship) => {
-        const relationshipType = relationship.relationshipType;
-        const ownerSide = relationship.ownerSide;
-        const relationshipName = relationship.relationshipName;
-        const relationshipFieldName = relationship.relationshipFieldName; _%>
-    <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'many-to-many' && ownerSide === true) || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
+  <%_ } _%>
+<%_ }); _%>
+<%_ relationships.forEach((relationship) => {
+    const ownerSide = relationship.ownerSide;
+    const relationshipName = relationship.relationshipName; _%>
+  <%_ if (relationship.relationshipManyToOne || (relationship.relationshipManyToMany && ownerSide) || (relationship.relationshipOneToOne && ownerSide)) { _%>
```

Comment:
```suggestion
  <%_ if (ownerSide) { _%>
```

relationshipManyToOne is implicit ownerSide and we started setting it at jhipster 7.

Reply:
What about oneToMany?